app/javascript/react/screens/App/Overview/components/Migrations/MigrationsInProgressCard.js

```diff
@@ -19,14 +19,138 @@ const MigrationsInProgressCard = ({ plan, allRequestsWithTasks, reloadCard, hand
   const requestsOfAssociatedPlan = allRequestsWithTasks.filter(request => request.source_id === plan.id);
   const mostRecentRequest = requestsOfAssociatedPlan.length > 0 && getMostRecentRequest(requestsOfAssociatedPlan);
 
+  // === TODO FIXME-- THE BELOW IS MOCK MUTATION CODE TO BE REMOVED ===
+  // This code:
+  // * sticks a fake active pre-playbook in the first migration on the screen
+  // * sticks a fake active post-playbook in the third migration on the screen
+  const dga = window.dangerousGlobalAccumulator;
+  const mockTasks = mostRecentRequest.miq_request_tasks.map((task, i) => ({
+    ...task,
+    options: {
+      ...task.options,
+      playbooks: {
+        pre: {
+          job_id: 4,
+          status: dga === 0 && i === 0 ? 'Active' : 'Succeeded',
+          last_task: 'Task name'
+        },
+        post: {
+          job_id: 5,
+          status: dga === 2 && i === 0 ? 'Active' : 'Succeeded',
+          last_task: 'Task name'
+        }
+      }
+    }
+  }));
+  window.dangerousGlobalAccumulator += 1;
+  const mostRecentTasks = mockTasks;
+  // ^^^ TODO FIXME-- THE ABOVE IS MOCK MUTATION CODE TO BE REMOVED ^^^
+  // We should remove the above code when the real API data is in place.
+
+  // const mostRecentTasks = mostRecentRequest.miq_request_tasks;
+
+  // TODO -- Much of the below 20 lines of sanity checking is probably redundant with Mike Ro's code.
+  // When we merge the two playbook-related branches together we should consolidate these kinds of things.
+
+  const hasNoPlaybooks = task => !task || !task.options || !task.options.playbooks;
+  const getActivePlaybook = task => {
+    if (hasNoPlaybooks(task)) return {};
+    const { pre, post } = task.options.playbooks;
+    if (pre.status === 'Active') return pre;
+    if (post.status === 'Active') return post;
+    return {};
+  };
+
+  const tasksWithActivePlaybooks = mostRecentTasks.filter(task => {
```

Comment:
I believe we should be using the `tasks[task.source_id]` task for analyzing playbooks. This ensures that we are not checking an old task from a previous migration attempt (that logic changed during the "Retry" rework). Will attempt to refactor this bit...